irohad/ametsuchi/impl/postgres_block_storage.hpp

```diff
@@ -8,24 +8,27 @@
 
 #include "ametsuchi/block_storage.hpp"
 
+#include "ametsuchi/impl/pool_wrapper.hpp"
 #include "ametsuchi/impl/soci_utils.hpp"
 #include "backend/protobuf/block.hpp"
+#include "backend/protobuf/proto_block_factory.hpp"
 #include "interfaces/iroha_internal/abstract_transport_factory.hpp"
 #include "logger/logger_fwd.hpp"
 
 namespace iroha {
   namespace ametsuchi {
     class PostgresBlockStorage : public BlockStorage {
      public:
-      using BlockTransportFactory =
-          shared_model::interface::AbstractTransportFactory<
-              shared_model::interface::Block,
-              shared_model::proto::Block::TransportType>;
+      using BlockTransportFactory = shared_model::proto::ProtoBlockFactory;
 
-      PostgresBlockStorage(soci::session &sql,
+      PostgresBlockStorage(std::shared_ptr<PoolWrapper> pool_wrapper,
                            std::shared_ptr<BlockTransportFactory> block_factory,
+                           std::string table,
+                           bool is_temporary,
```

Comment:
And related change here, maybe instead create another class `PostgresTemporaryBlockStorage`, inherit from `PostgresBlockStorage` and override destructor?